src/components/SelectionListRadio/BaseSelectionListRadio.js

```diff
@@ -0,0 +1,378 @@
+    textInputMaxLength: PropTypes.number,
+    onChangeText: PropTypes.func,
+    keyboardType: PropTypes.string,
+    initiallyFocusedOptionKey: PropTypes.string,
+    shouldDelayFocus: PropTypes.bool,
+    onScroll: PropTypes.func,
+    onScrollBeginDrag: PropTypes.func,
+    headerMessage: PropTypes.string,
+};
+
+const defaultProps = {
+    onSelectRow: () => {},
+    textInputLabel: '',
+    textInputPlaceholder: '',
+    textInputValue: '',
+    textInputMaxLength: undefined,
+    keyboardType: CONST.KEYBOARD_TYPE.DEFAULT,
+    onChangeText: () => {},
+    initiallyFocusedOptionKey: '',
+    shouldDelayFocus: false,
+    onScroll: () => {},
+    onScrollBeginDrag: () => {},
+    headerMessage: '',
+};
+
+function SelectionListRadio(props) {
+    const listRef = useRef(null);
+    const pressableRef = useRef(null);
+    const textInputRef = useRef(null);
+    const shouldShowTextInput = Boolean(props.textInputLabel);
+
+    const flattenedSections = useMemo(() => {
+        const allOptions = [];
+
+        const disabledOptionsIndexes = [];
+        let disabledIndex = 0;
+
+        let offset = 0;
+        const itemLayouts = [{length: 0, offset}];
+
+        _.each(props.sections, (section, sectionIndex) => {
+            // We're not rendering any section header, but we need to push to the array
+            // because React Native accounts for it in getItemLayout
+            const sectionHeaderHeight = 0;
+            itemLayouts.push({length: sectionHeaderHeight, offset});
+            offset += sectionHeaderHeight;
+
+            _.each(section.data, (option, optionIndex) => {
+                // Add item to the general flattened array
+                allOptions.push({
+                    ...option,
+                    sectionIndex,
+                    index: optionIndex,
+                });
+
+                // If disabled, add to the disabled indexes array
+                if (section.isDisabled || option.isDisabled) {
+                    disabledOptionsIndexes.push(disabledIndex);
+                }
+                disabledIndex += 1;
+
+                // Account for the height of the item in getItemLayout
+                const fullItemHeight = variables.optionRowHeight;
+                itemLayouts.push({length: fullItemHeight, offset});
+                offset += fullItemHeight;
+            });
+
+            // We're not rendering any section footer, but we need to push to the array
+            // because React Native accounts for it in getItemLayout
+            itemLayouts.push({length: 0, offset});
+        });
+
+        // We're not rendering the list footer, but we need to push to the array
+        // because React Native accounts for it in getItemLayout
+        itemLayouts.push({length: 0, offset});
+
+        return {
+            allOptions,
+            disabledOptionsIndexes,
+            itemLayouts,
+        };
+    }, [props.sections]);
+
+    const [focusedIndex, setFocusedIndex] = useState(() => {
```

Comment:
I propose to change the order of hooks:
1. useStates
2. useEffects
3. functions and useCallbacks
4. consts and useMemos
of course, if the order of declared variables allow that and does not depend on each other

Reply:
I tried but there are a few things that depend on each other. This is the best I came up with:
1. consts, refs
2. functions (1 const and 1 useState had to be in the middle)
3. effects